src/Datadog.AutoInstrumentation.NativeLoader/CMakeLists.txt

```diff
@@ -0,0 +1,173 @@
+
+# Set output folders
+SET(OUTPUT_BIN_DIR ${CMAKE_BINARY_DIR}/bin)
+SET(OUTPUT_TMP_DIR ${CMAKE_BINARY_DIR}/tmp.${CMAKE_SYSTEM_NAME}_${CMAKE_SYSTEM_PROCESSOR})
+SET(OUTPUT_DEPS_DIR ${CMAKE_BINARY_DIR}/deps)
+FILE(MAKE_DIRECTORY ${OUTPUT_BIN_DIR})
+FILE(MAKE_DIRECTORY ${OUTPUT_TMP_DIR})
+FILE(MAKE_DIRECTORY ${OUTPUT_DEPS_DIR})
+FILE(COPY ${CMAKE_CURRENT_SOURCE_DIR}/loader.conf DESTINATION ${CMAKE_BINARY_DIR}/bin)
+
+
+SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${OUTPUT_BIN_DIR})
+SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${OUTPUT_BIN_DIR})
+SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${OUTPUT_BIN_DIR})
+
+# ******************************************************
+# Compiler options
+# ******************************************************
+
+# Sets the compiler
+if(ISLINUX)
+    SET (CMAKE_C_COMPILER   /usr/bin/clang)
+    SET (CMAKE_CXX_COMPILER /usr/bin/clang++)
+endif()
+
+# Sets compiler options
+add_compile_options(-std=c++17 -fPIC -fms-extensions)
+add_compile_options(-DPAL_STDCPP_COMPAT -DPLATFORM_UNIX -DUNICODE)
+add_compile_options(-Wno-invalid-noreturn -Wno-macro-redefined)
+if (ISMACOS)
+    add_compile_options(-stdlib=libc++ -DMACOS -Wno-pragma-pack)
+elseif(ISLINUX)
+    add_compile_options(-stdlib=libstdc++ -DLINUX -Wno-pragmas)
+endif()
+if (BIT64)
+    add_compile_options(-DBIT64 -DHOST_64BIT)
+else()
+    add_compile_options(-DBIT86 -DHOST_X86)
+endif()
+if (ISAMD64)
+    add_compile_options(-DAMD64)
+elseif (ISX86)
+    add_compile_options(-DX86)
+elseif (ISARM64)
+    add_compile_options(-DARM64)
+elseif (ISARM)
+    add_compile_options(-DARM)
+endif()
+
+# ******************************************************
+# Suppress Warning on MacOS
+# ******************************************************
+
+# Only necessary with cmake 3.19.x on macos
+# See https://stackoverflow.com/questions/4929255/building-static-libraries-on-mac-using-cmake-and-gcc#answer-4953904
+
+if (ISMACOS)
+    SET(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
+    SET(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> Scr <TARGET> <LINK_FLAGS> <OBJECTS>")
+    SET(CMAKE_C_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+    SET(CMAKE_CXX_ARCHIVE_FINISH "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+endif()
+
+# ******************************************************
+# Define target
+# ******************************************************
+add_library("Datadog.AutoInstrumentation.NativeLoader" SHARED
+        cor_profiler.cpp
+        cor_profiler_class_factory.cpp
+        dllmain.cpp
+        dynamic_dispatcher.cpp
+        dynamic_instance.cpp
+        miniutf.cpp
+        pal.cpp
+        string.cpp
+        ../Datadog.Trace.ClrProfiler.Native/lib/coreclr/src/pal/prebuilt/idl/corprof_i.cpp
+        ${GENERATED_OBJ_FILES}
+        )
+
+set_target_properties("Datadog.AutoInstrumentation.NativeLoader" PROPERTIES PREFIX "")
+
+# Define directories includes
+target_include_directories("Datadog.AutoInstrumentation.NativeLoader"
```

Comment:
@macrogreg in your directory structure plan you plan to move these out into a different folder instead of Datadog.Trace.ClrProfiler.Native, right?

Reply:
FWIW, I think so, we will need to move shared components to a common folder. This should include CorProfiler dependencies like CLR headers. But also common code, like logging, string handling and PAL methods. I didn't do that here because is not the goal of this PR.